Fixed #336. Classpath handling BasicDB from source

kruthar opened the pull request:
To accomplish this, just need to alter the classpath logic when given 'basic' as the DB. It looked like you just needed a built core jar to run the BasicDB from source.

get_classpath_from_maven(database) now puts the database directly into maven command as opposed to appending '-bending'. This allowed all logic to happen in main().

In main() now using the 'core' dir and the 'core' project if using the 'basic' database. Otherwise things happen as normal.

busbey commented:
thanks for working on this!

please amend the commit message to start with an indicator of what section has changed and a brief summation.

e.g:

```commit                                                               
[scripts] allow running against basic DB from source.                   
                                                                        
get_classpath_from_maven(database) now does not alter the project name  
before giving it to Maven. main() now calculates the db_dir with 'core' 
and uses the core project for Maven, otherwise things work as they did  
before.                                                                 
                                                                        
fixes #336                                                              
```

busbey commented on bin/ycsb:
```diff
@@ -140,9 +140,9 @@ def is_distribution():
+        debug("Running 'mvn -pl com.yahoo.ycsb:"+database+" -am package -DskipTests "
               "dependency:build-classpath -DincludeScope=compile -Dmdep.outputFilterFile=true'")
-        mvn_output = subprocess.check_output(["mvn", "-pl", "com.yahoo.ycsb:"+database+"-binding",
+        mvn_output = subprocess.check_output(["mvn", "-pl", "com.yahoo.ycsb:" + database,
```
won't this fail to build the correct module for any of the datastore specific bindings? e.g., if I request hbase10 the name of that module is 'hbase10-binding'

busbey replied:
oh I see, we'd pass in 'hbase10-binding' when we call the method now.

could you update get_classpath_from_maven to specify that the parameter is 'module' now rather than 'database'? that would make this much clearer.

kruthar commented:
Okie Dokie. After learning a little bit of what to do and not to do with altering git history...I updated the commit message and added some self documenting refactor changes to get_classpath_from_maven()

busbey commented:
thanks for the contribution!
